cmd/util/ledger/migrations/cadence_values_migration.go

```diff
@@ -220,23 +252,59 @@ func NewCadence1CapabilityValueMigrator(
 				},
 			}
 		},
+		errorMessageHandler: errorMessageHandler,
+	}
+}
+
+// errorMessageHandler formats error messages from errors.
+// It only reports program loading errors once.
+type errorMessageHandler struct {
+	// common.Location -> struct{}
+	reportedProgramLoadingErrors sync.Map
+}
+
+func (t *errorMessageHandler) FormatError(err error) string {
+	var message string
+
+	// Only report program loading errors once,
+	// omit full error message for subsequent occurrences
+
+	var programLoadingError environment.ProgramLoadingError
+	if errors.As(err, &programLoadingError) {
+		location := programLoadingError.Location
+		_, ok := t.reportedProgramLoadingErrors.LoadOrStore(location, struct{}{})
+		if ok {
+			message = "error getting program"
```

Comment:
can just return from here
```suggestion
			return "error getting program"
```

Reply:
Simplified in [388a112](https://github.com/onflow/flow-go/pull/5388/commits/388a11294e702b48d13f62c769a03b4006b0658f)